project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/MinimizationPluginStateSerializer.kt

```diff
@@ -0,0 +1,40 @@
+package org.plan.research.minimization.plugin.settings
+
+import org.plan.research.minimization.plugin.services.MinimizationPluginSettings
+
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.JDOMUtil
+import com.intellij.util.xmlb.XmlSerializer
+import org.jdom.Element
+
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Paths
+
+fun saveStateToFile(project: Project, filePath: String) {
+    val state = project.service<MinimizationPluginSettings>().state
+    val element = serializeState(state)
+    val xmlString = JDOMUtil.writeElement(element, System.lineSeparator())
+
+    Files.newBufferedWriter(Paths.get(filePath)).use { writer ->
+        writer.write(xmlString)
+    }
```

Comment:
There is a [function](https://github.com/JetBrains/intellij-community/blob/0f010147bcdba4ff78f6fe8273866cd0c7f22755/platform/util/src/com/intellij/openapi/util/JDOMUtil.java#L473) for this